File: src/Omnia.ProcessManagement.Models/Processes/InternalProcessItem.cs
using Omnia.Fx.Models.Language;
using System;
using System.Collections.Generic;
using System.Text;

namespace Omnia.ProcessManagement.Models.Processes
{
    public class InternalProcessItem : ProcessItem
    {
        public override ProcessItemTypes Type => ProcessItemTypes.Internal;
        public List<ProcessItem> ProcessItems { get; set; }
        public Guid ProcessDataId { get; set; }
    }
}

File: src/Omnia.ProcessManagement.Web/client/fx/models/data/processes/InternalProcessItem.ts
import { GuidValue, LanguageTag } from '@omnia/fx-models';
import { ProcessItem, ProcessItemTypes } from './ProcessItem';

export interface InternalProcessItem extends ProcessItem {
    type: ProcessItemTypes.Internal;
    children: Array<ProcessItem>;
    processMetadataId: GuidValue;
}